src/containers/IntentHandler.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react'
+import { cozyConnect } from 'redux-cozy-client'
+import { connect } from 'react-redux'
+
+import Loading from '../components/Loading'
+import IntentService from './IntentService'
+
+import { initializeRegistry } from '../ducks/registry'
+import { fetchAccounts } from '../ducks/accounts'
+import { fetchKonnectorJobs } from '../ducks/jobs'
+import { fetchKonnectors } from '../ducks/konnectors'
+import { fetchTriggers } from '../ducks/triggers'
+
+class IntentHandler extends Component {
+  constructor(props, context) {
+    super(props, context)
+    this.store = context.store
+
+    this.state = {
+      isInitializing: true
+    }
+
+    props.initializeRegistry(props.initKonnectors)
+
+    this.store
+      .createIntentService()
+      .then(service => {
+        this.setState({
+          isInitializing: false,
+          service: service
+        })
+      })
+      .catch(error => {
+        this.setState({
+          isInitializing: false,
+          error: {
+            message: 'intent.service.error.initialization',
+            reason: error.message
+          }
+        })
+      })
+  }
+
+  terminate(account) {
+    const { service } = this.state
+    service.terminate(account)
+  }
+
+  cancel() {
+    const { service } = this.state
+    service.cancel()
+  }
+
+  handleError(error) {
+    this.setState({
+      error: {
+        message: 'intent.service.error.creation',
+        reason: error.message
+      }
+    })
+
+    throw error
+  }
+
+  render() {
+    // const { data } = this.props
+    const { appData, accounts, konnectors, triggers } = this.props
+    const { error, service } = this.state
+
+    const { t } = this.context
+
+    let { isInitializing } = this.state
+
+    isInitializing =
+      isInitializing ||
+      [accounts, konnectors, triggers].find(collection =>
+        ['pending', 'loading'].includes(collection.fetchStatus)
```

Comment:
TIL [`Array.includes`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/includes)

Reply:
Welcome to the plateformers